sdk/core/System.ClientModel/src/Internal/ArrayBackedPropertyBag.cs

```diff
@@ -0,0 +1,308 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace System.ClientModel.Internal;
+
+/// <summary>
+/// A property bag which is optimized for storage of a small number of items.
+/// If the item count is less than 2, there are no allocations. Any additional items are stored in an array which will grow as needed.
+/// MUST be passed by ref only.
+/// </summary>
+internal struct ArrayBackedPropertyBag<TKey, TValue> where TKey : struct, IEquatable<TKey>
+{
+    private Kvp _first;
+    private Kvp _second;
+    private Kvp[]? _rest;
+    private int _count;
+#if DEBUG
+    private bool _disposed;
+#endif
+    private readonly struct Kvp
```

Comment:
should we use a value tuple instead? i.e. (TKey key, TValue value). Is there a benefit to using this custom struct?

Reply:
This is a type we copied in from Azure.Core, which I suspect may have originally come from the dotnet/runtime repo.  I don't have strong feelings about it or changes to it, but unless it's high prio, I'd rather not spend a lot of time on it.  Let me know how strongly you feel about changes here, @KrzysztofCwalina.

Reply:
Value tuples have almost exact APIs (and almost same layout). The change should take 5 minutes, e.g. replace "private Kvp _first" with "private (Tkey Key, TValue Value) _first". I think it will simplify this file. If it's longer than 5 mins, feel free to punt.

Reply:
Sounds good!  I'll change it in both places to keep the files consistent, assuming it's a quick a fix as you expect.  Thanks!